### PR TITLE
Simple CLP for safely converting from IntervalList to BED.

### PR DESCRIPTION
**Pull request by tfenne**

### CODE REVIEW

File: src/java/picard/util/IntervalListToBed.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    public File INPUT;
+
+    @Option(shortName=StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc="Output BED file.")
+    public File OUTPUT;
+
+    @Option(doc="The score, between 0-1000, to output for each interval in the BED file.")
+    public int SCORE = 500;
+
+    @Option(doc="If true, sort the interval list prior to outputting as BED file.")
+    public boolean SORT= true;
```
Comment (nh13) [inline src/java/picard/util/IntervalListToBed.java]: `SORT=` -> `SORT =`

File: src/tests/java/picard/util/IntervalListToBedTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        final IntervalListToBed program = new IntervalListToBed();
+        final File tmp = File.createTempFile("interval_list_to_bed_test_output", ".bed");
+        tmp.deleteOnExit();
+
+        final String[] args = {
+                "INPUT=" + INTERVAL_LIST.getAbsolutePath(),
+                "OUTPUT=" + tmp.getAbsolutePath(),
+                "SCORE=333"
+        };
+        program.instanceMain(args);
```
Comment (nh13) [inline src/tests/java/picard/util/IntervalListToBedTest.java]: why not use `CommandLineProgramTest`?

Comment (tfenne) [inline src/tests/java/picard/util/IntervalListToBedTest.java]: I modeled this a little after BedToIntervalList test which doesn't either.  But now that I look at CommandLineProgramTest it doesn't seem to do much.

**Comment by tfenne** [comment]
@ktibbett ok to merge this?

File: src/java/picard/util/IntervalListToBed.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+)
+public class IntervalListToBed extends CommandLineProgram {
+    @Option(shortName=StandardOptionDefinitions.INPUT_SHORT_NAME, doc="Input IntervalList file.")
+    public File INPUT;
+
+    @Option(shortName=StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc="Output BED file.")
+    public File OUTPUT;
+
+    @Option(doc="The score, between 0-1000, to output for each interval in the BED file.")
+    public int SCORE = 500;
```
Comment (yfarjoun) [inline src/java/picard/util/IntervalListToBed.java]: what is the "score" for?

File: testdata/picard/util/interval_list_to_bed_test.interval_list [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+@SQ	SN:chr2	LN:1000000
+@SQ	SN:chr3	LN:1000000
+@SQ	SN:chr4	LN:1000000
+@SQ	SN:chr5	LN:1000000
+chr1	1	50	+	target_at_chrom_start
+chr1	100	110	+	target_1
+chr2	200	220	-	target_2
+chr3	300	330	+	target_3
+chr4	400	440	+	target_4
+chr5	500	550	-	target_5
```
Comment (yfarjoun) [inline testdata/picard/util/interval_list_to_bed_test.interval_list]: please include an example without a name, assuming that it's a legal entry.

Comment (yfarjoun) [inline testdata/picard/util/interval_list_to_bed_test.interval_list]: what about empty intervals? please include a zero-length interval.

Comment (yfarjoun) [inline testdata/picard/util/interval_list_to_bed_test.interval_list]: nevermind abut the name. name is required.

File: src/java/picard/util/IntervalListToBed.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+        IntervalList intervals = IntervalList.fromFile(INPUT);
+        if (SORT) intervals = intervals.sorted();
+
+        try {
+            final BufferedWriter out = IOUtil.openFileForBufferedWriting(OUTPUT);
+            for (final Interval i : intervals) {
+                final String strand = i.isNegativeStrand() ? "-" : "+";
+                final List<?> fields = CollectionUtil.makeList(i.getContig(), i.getStart()-1, i.getEnd(), i.getName(), SCORE, strand);
+                out.append(fields.stream().map(String::valueOf).collect(Collectors.joining("\t")));
```
Comment (yfarjoun) [inline src/java/picard/util/IntervalListToBed.java]: I like this java8 sugar...:-)

**Comment by yfarjoun** [comment]
minor questions/requests. then :+1:  from me.

File: src/java/picard/util/IntervalListToBed.java (hunk 3 above)
Comment (tfenne) [inline src/java/picard/util/IntervalListToBed.java]: It's simply field #5 in the BED file format.  The UCSC browser uses it for coloring entries.  But you have to put something there if you want to include further fields, and strand is in field 6.

**Comment by tfenne** [comment]
Thanks @yfarjoun - I added the zero-length interval to the test data.  Will merge once travis passes again.

**Pull request merged**